From c5f12d150d8f4b9b920f6e9511d5629b81333b2b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Markus=20M=C3=BCtzel?= <markus.muetzel@gmx.de>
Date: Fri, 17 May 2024 19:47:39 +0200
Subject: [PATCH] Keep original LDFLAGS.

* oct/Makefile_unix: Append to original LDFLAGS. Keep setting LFLAGS for older
versions of Octave (prior to version 7).
---
 oct/Makefile_unix | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/oct/Makefile_unix b/oct/Makefile_unix
index de2bce8e..7b224a03 100644
--- a/oct/Makefile_unix
+++ b/oct/Makefile_unix
@@ -34,11 +34,11 @@ FFTW_LIBS := $(shell $(MKOCTFILE) -p FFTW_LIBS)
 endif
 
 export CXXFLAGS := $(shell $(MKOCTFILE) -p CXXFLAGS) -std=gnu++11 -Wall -DLTFAT_LARGEARRAYS $(OPTCXXFLAGS)
-# export is necessary, otherwise LFLAGS won't have any effect
+# export is necessary, otherwise LDFLAGS won't have any effect
 # at least on Windows and on Mac
-export LDFLAGS := -L$(shell $(MKOCTFILE) -p LIBDIR) $(FLIBS) $(LAPACK_LIBS) $(BLAS_LIBS) $(FFTW_LIBS) 
-undefine LFLAGS
-unexport LFLAGS
+export LDFLAGS := $(shell $(MKOCTFILE) -p LDFLAGS) $(FLIBS) $(LAPACK_LIBS) $(BLAS_LIBS) $(FFTW_LIBS)
+# also export as LFLAGS for Octave prior to version 7
+export LFLAGS := $(shell $(MKOCTFILE) -p LFLAGS) $(FLIBS) $(LAPACK_LIBS) $(BLAS_LIBS) $(FFTW_LIBS)
 
 MKOCTFILEFLAGS = -I../src/modules/libltfat/include -DNDEBUG -L../lib -lltfat
 
-- 
2.44.0.windows.1

